src/norm_oracles.jl

```diff
@@ -360,3 +360,61 @@ function compute_extreme_point(lmo::EllipsoidLMO, direction; v=nothing, kwargs..
     mul!(v, I, lmo.buffer, -scaling, true)
     return v
 end
+
+
+"""
+    OrderWeightNormLMO(weights,radius)
+    
+LMO with feasible set being the atomic ordered weighted l1 norm: https://arxiv.org/pdf/1409.4271
+
+```
+C = {x ∈ R^n, Ω_w(x) ≤ R} 
+```
+
+"""
+struct OrderWeightNormLMO{W,R,P,B,D} <: LinearMinimizationOracle
+    weights::W
+    radius::R
+    perm_indices::P
+    mat_B::B
+    direction_abs::D
+end
+
+function OrderWeightNormLMO(weights, radius)
+    N = length(weights)
+    mat = zeros(N,N)
+    s = zero(eltype(weights))
+    for i in 1:N
+        s += weights[i]
+        for j in 1:i
+            mat[i,j] = 1 / s
+        end
+    end
+    perm_indices = sortperm(weights,rev=true)
+    direction_abs = similar(weights)
+    return OrderWeightNormLMO(weights, radius, perm_indices, mat, direction_abs)
+end
+
+function compute_extreme_point(
+    lmo::OrderWeightNormLMO,
+    direction::M;
+    v=similar(direction),
+    kwargs...,
+) where {M}
+    for i in eachindex(direction)
+        lmo.direction_abs[i] = abs(direction[i])
+    end
+    perm_grad = sortperm(lmo.direction_abs,rev=true)
+    max = 0
+    ind = 1
+    N = length(lmo.weights)
+    for i in 1:N
+        scal = dot(lmo.mat_B[i,:],lmo.direction_abs[perm_grad])
+        if(scal > max)
+            max = scal
+            ind = i
+        end
+    end
+    v = (lmo.radius).*sign.(-1*direction).*((lmo.mat_B[ind,:])[sortperm(perm_grad)])
```

Comment:
you are redefining `v` here, you should probably use `v=nothing` in the keyword arguments then